plugins/magic.py

```diff
@@ -1,39 +1,67 @@
-"""!magic <search term> return the first magic card with name containing <search term>"""
+"""!magic <search term> return magic card specified by the search term"""
 
+import math
 import re
 import random
 from urllib import quote
 
 import requests
 
-def magic(searchterm):
-    quotedsearchterm = quote(searchterm)
-    url = "https://api.deckbrew.com/mtg/cards?name={}"
-    url = url.format(quotedsearchterm)
-    results = requests.get(url).json()
+filterlist = ["type","supertype","subtype","name","oracle","set","rarity","set","color","multicolor","m","format","status"]
 
-    if not results:
+def search(searchterm):
+    filters = getfilters(searchterm)
+    if filters:
+        search = reduce(lambda string, term: string + ("" if not string else "&") + "{0}={1}".format(term[0].strip(),quote(term[1].strip())), filters, "")
+        url = "https://api.deckbrew.com/mtg/cards?{}".format(search)
+        result = getresult(url)
+    else:
+        #if no filters found we default to name search
+        url = "https://api.deckbrew.com/mtg/cards?name={}".format(quote(searchterm))
+        results = requests.get(url).json()
```

Comment:
It could be paginated.  Will need a slight rework to check all pages for exact match.